From ccbeb1cc252090eda27d1642135856c654ac46b7 Mon Sep 17 00:00:00 2001
From: Fred Fredburger <fredfredburger86@protonmail.com>
Date: Thu, 28 May 2020 01:16:25 +0300
Subject: [PATCH 2/2] status2d tags: ^v^ - save fg/bg color, ^t^ - restore
 fg/bg color

---
 dwm.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dwm.c b/dwm.c
index 6c16cf2..58391f2 100644
--- a/dwm.c
+++ b/dwm.c
@@ -779,6 +779,7 @@ drawstatusbar(Monitor *m, int bh, char* stext) {
 	short isCode = 0;
 	char *text;
 	char *p;
+	Clr oldbg, oldfg;
 
 	len = strlen(stext) + 1 ;
 	if (!(text = (char*) malloc(sizeof(char)*len)))
@@ -854,6 +855,12 @@ drawstatusbar(Monitor *m, int bh, char* stext) {
 					swp = drw->scheme[ColFg];
 					drw->scheme[ColFg] = drw->scheme[ColBg];
 					drw->scheme[ColBg] = swp;
+				} else if (text[i] == 'v') {
+					oldfg = drw->scheme[ColFg];
+					oldbg = drw->scheme[ColBg];
+				} else if (text[i] == 't') {
+					drw->scheme[ColFg] = oldfg;
+					drw->scheme[ColBg] = oldbg;
 				} else if (text[i] == 'r') {
 					int rx = atoi(text + ++i);
 					while (text[++i] != ',');
-- 
2.26.2

